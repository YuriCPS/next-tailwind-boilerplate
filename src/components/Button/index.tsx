export type ButtonProps = {
  href: string
  text: string
  color: string
  size: string
};

interface Styles {
  bg: { [key: string]: string }
  size: { [key: string]: string }
  common: string,
}

const style: Styles = {
  bg: {
    amber: 'bg-amber-400 hover:bg-amber-500',
    blue: 'bg-blue-600 hover:bg-blue-700 text-white',
    green: 'bg-green-600 hover:bg-green-700 text-white',
    neutral: 'bg-neutral-700 hover:bg-neutral-900 text-white',
    red: 'bg-red-600 hover:bg-red-700 text-white',
    violet: 'bg-violet-600 hover:bg-violet-800 text-white',
  },
  size: {
    small: 'text-sm',
    medium: 'text-md',
    large: 'text-xl',
  },
  common: 'hover:scale-110 m-2 py-2 px-4 rounded-md font-medium',
};

const Button = ({
  href, text, color, size,
}: ButtonProps) => (
  <a href={href}>
    <button
      type="button"
      className={`${style.bg[color]} ${style.size[size]} ${style.common}`}
    >
      {text}
    </button>
  </a>
);

export default Button;
