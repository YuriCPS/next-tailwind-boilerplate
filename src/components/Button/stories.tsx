import { Story, Meta } from '@storybook/react';
import Button, { ButtonProps } from '.';

export default {
  title: 'Button',
  component: Button,
} as Meta;

export const Amber: Story<ButtonProps> = (args) => <Button {...args} />;

Amber.args = {
  href: '#',
  text: 'Button',
  color: 'amber',
  size: 'medium',
};

export const Blue: Story<ButtonProps> = (args) => <Button {...args} />;

Blue.args = {
  href: '#',
  text: 'Button',
  color: 'blue',
  size: 'medium',
};

export const Green : Story<ButtonProps> = (args) => <Button {...args} />;

Green.args = {
  href: '#',
  text: 'Button',
  color: 'green',
  size: 'medium',
};

export const Neutral : Story<ButtonProps> = (args) => <Button {...args} />;

Neutral.args = {
  href: '#',
  text: 'Button',
  color: 'neutral',
  size: 'medium',
};

export const Red : Story<ButtonProps> = (args) => <Button {...args} />;

Red.args = {
  href: '#',
  text: 'Button',
  color: 'red',
  size: 'medium',
};

export const Violet : Story<ButtonProps> = (args) => <Button {...args} />;

Violet.args = {
  href: '#',
  text: 'Button',
  color: 'violet',
  size: 'medium',
};
