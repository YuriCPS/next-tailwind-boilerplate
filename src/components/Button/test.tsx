import { render, screen } from '@testing-library/react';

import Button from '.';

describe('<Button />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Button
        href="href"
        text="Button"
        color="blue"
        size="medium"
      />,
    );

    expect(screen.getByRole('button', { name: /Button/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
