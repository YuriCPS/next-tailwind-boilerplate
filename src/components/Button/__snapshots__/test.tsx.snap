// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
<a
  href="href"
>
  <button
    class="p-2 m-2 rounded-md bg-amber-400 hover:scale-110 hover:bg-amber-500"
    type="button"
  >
    title
  </button>
</a>
`;
